proxy/proxy_test.go

```diff
@@ -73,6 +75,87 @@ type listener struct {
 	lastConn chan net.Conn
 }
 
+type blockableBuffer struct {
+	inner *bytes.Buffer
+	m     sync.RWMutex
+}
+
+func (w *blockableBuffer) Write(p []byte) (int, error) {
+	w.Lock()
+	defer w.Unlock()
+
+	return w.inner.Write(p)
+}
+
+func (w *blockableBuffer) String() string {
+	w.RLock()
+	defer w.RUnlock()
+
+	return w.inner.String()
+}
+
+func (w *blockableBuffer) Lock() {
+	w.m.Lock()
+}
+
+func (w *blockableBuffer) Unlock() {
+	w.m.Unlock()
+}
+
+func (w *blockableBuffer) RLock() {
+	w.m.RLock()
+}
+
+func (w *blockableBuffer) RUnlock() {
+	w.m.RUnlock()
+}
+
+type testLog struct {
+	buf      *blockableBuffer
+	oldOut   io.Writer
+	oldLevel log.Level
+}
+
+func NewTestLog() *testLog {
+	oldOut := os.Stderr
+	oldLevel := log.GetLevel()
+	buf := &blockableBuffer{inner: &bytes.Buffer{}, m: sync.RWMutex{}}
```

Comment:
you can omit `m: sync.RWMutex{}` and the innre as well if you change it to the value instead of the pointer